src/openrct2/command_line/SpriteCommands.cpp

```diff
@@ -14,9 +14,29 @@
 
 using namespace OpenRCT2;
 
-#define SZ_DEFAULT "default"
-#define SZ_CLOSEST "closest"
-#define SZ_DITHERING "dithering"
+template<size_t... len> constexpr auto concat(const char (&... strings)[len])
+{
+    constexpr size_t n = (... + len) - sizeof...(len);
+    std::array<char, n + 1> result = {};
+    result[n] = '\0';
+
+    size_t i = 0;
+    for (const char* src : { strings... })
+    {
+        for (; *src != '\0'; src++, i++)
+        {
+            result[i] = *src;
+        }
+    }
+
+    return result;
+}
+
+constexpr const char kSzDefault[] = "default";
+constexpr const char kSzClosest[] = "closest";
+constexpr const char kSzDithering[] = "dithering";
+static constexpr auto _commandLineOptionDefintion = concat(
+    "the type of sprite conversion <", kSzDefault, "|", kSzClosest, "|", kSzDithering, ">");
```

Comment:
I think this is overengineering things. Considering these constants are only for use in this unit, couldn't we use `std::string` and its concatenation operations instead?

Reply:
My understanding is that while you can declare an std::string with static constexpr, you cannot then concatenate those constants together, so the example below doesn't work.

```
static constexpr std::string foo = "whatever";
static constexpr std::string bar = "something else";
static constexpr std::string something = foo + bar;
```

I certainly agree that this may be overengineering though. The only reason to use this is if we really care about replacing the defines, which I'm not sure if we do.

Reply:
I have to agree with @AaronVanGeffen. This area of code is one of the oldest parts, its also not the greatest design. We have to come up with a better way of handling this at some point, but its also not exactly an area that has high priority as it just works for the time being.

Reply:
Sounds good to me. I still think that the enum change is reasonable though. Should I just close this PR and save it for later or revert the SpriteCommand.cpp changes and keep the PR for the enums?

Reply:
The enum change is fine.